configtxt/boot: Update arm_64bit flag (#3903)
raspberrypi/documentation: documentation/asciidoc/computers/config_txt/boot.adoc

=== `arm_64bit`

> NOTE: This flag has been removed from Raspberry Pi 5, as Raspberry Pi 5 only supports a 64-bit kernel.

Reviewer: ```suggestion
```

I suggest removing this because notes are disruptive to section flow and I think I've found a better way to convey this information in the text below. In particular, while calling out this specific model _might_ be helpful right now, it's not very futureproof: as we release more models, we'll have to expand this note to a massive and unhelpful size. Better to clarify in the meat and potatoes of this section, I think.

If set to 1, the kernel will be started in 64-bit mode. Setting to 0 selects 32-bit mode.

In 64-bit mode, the firmware will choose an appropriate kernel (e.g. `kernel8.img`), unless there is an explicit `kernel` option defined, in which case that is used instead.

Defaults to 1 on Raspberry Pi 4, 400 and Compute Module 4, 4S platforms. Defaults to 0 on all other platforms. However, if the name given in an explicit `kernel` option matches one of the known kernels then `arm_64bit` will be set accordingly.

> NOTE: 64-bit kernels may be uncompressed image files or a gzip archive of an image (which can still be called kernel8.img; the bootloader will recognize the archive from the signature bytes at the beginning). The 64-bit kernel will only work on the Raspberry Pi 3, 3+, 4, 400, Zero 2 W and 2B rev 1.2, and Raspberry Pi Compute Modules 3, 3+ and 4.

Reviewer: ```suggestion
64-bit kernels come in the following forms:

* uncompressed image files
* gzip archives of an image

Both forms may use the `img` file extension; the bootloader recognizes archives using signature bytes at the start of the file.

The following Raspberry Pi models support this flag:

* 2B rev 1.2
* 3
* 3+
* 4
* 400
* Zero 2 W
* Compute Module 3
* Compute Module 3+
* Compute Module 4

Later models, such as Raspberry Pi 5, _only_ support the 64-bit kernel. Such models do not support this flag.
```

=== `arm_64bit`

If set to 1, the kernel will be started in 64-bit mode. Setting to 0 selects 32-bit mode.

In 64-bit mode, the firmware will choose an appropriate kernel (e.g. `kernel8.img`), unless there is an explicit `kernel` option defined, in which case that is used instead.

Defaults to 1 on Raspberry Pi 4, 400 and Compute Module 4, 4S platforms. Defaults to 0 on all other platforms. However, if the name given in an explicit `kernel` option matches one of the known kernels then `arm_64bit` will be set accordingly.

64-bit kernels come in the following forms:

* uncompressed image files

> * gzip archives of an image

Reviewer 2: I wonder if we should remove the plural from these two lines, given that you'll only ever be using one kernel at any given time? And IMHO "archive" usually implies that there's more than one file inside the archive, so perhaps this could be simplified to? (so "uncompressed image file" and "gzip-compressed image file" ?)

* gzip archives of an image

Both forms may use the `img` file extension; the bootloader recognizes archives using signature bytes at the start of the file.

The following Raspberry Pi models support this flag:

* 2B rev 1.2
* 3
* 3+
* 4
* 400
* Zero 2 W
* Compute Module 3
* Compute Module 3+

> * Compute Module 4

Reviewer: ```suggestion
* Compute Module 4
* Compute Module 4S
```
I think the original writer missed this, or forgot to update it when we released 4S and added it above.

In 64-bit mode, the firmware will choose an appropriate kernel (e.g. `kernel8.img`), unless there is an explicit `kernel` option defined, in which case that is used instead.

> Defaults to 1 on Raspberry Pi 4, 5, 400 and Compute Module 4, 4S platforms. Defaults to 0 on all other platforms. However, if the name given in an explicit `kernel` option matches one of the known kernels then `arm_64bit` will be set accordingly.

Reviewer 3: I'd add CM5 as well - everyone knows its coming.

Author: I try to avoid mentioning any potential products ahead of a formal announcement, especially if I've structured the text such that any new products could be trivially included in the line.

Reviewer: Agree with Tom in principle. But for my own edification: I assumed that CM5 wouldn't support 32-bit mode at all, just like Pi 5. So I don't think we need to add it here at release. Is that correct, @JamesH65 ? Or will CM5 support 32-bit mode?

Reviewer 2: I believe it's actually the SoC (and not the specific Model) that determines if it's capable of booting 32-bit kernel only (BCM2835, BCM2836), either 32- or 64-bit kernel (BCM2837, RP3A0, BCM2711), or 64-bit kernel only (BCM2712) ?

But (in general) we explicitly list the Models in documentation rather than SoCs, because people find it easier to identify their Model than to identify their SoC :socks: :wink:
